polaris-core/src/main/java/org/apache/polaris/core/secrets/UserSecretReference.java

```diff
@@ -88,8 +88,7 @@ public UserSecretReference(
    */
   @JsonIgnore
   public String getUserSecretManagerTypeFromUrn() {
-    // TODO: Add better/standardized parsing and validation of URN syntax
-    return urn.split(":")[2];
+    return UserSecretReferenceUrnHelper.getSecretManagerType(urn);
```

Comment:
It might be nice just to offer something like `UserSecretReference usr = UserSecretReference.fromUrn(...)`

Reply:
At the moment, we don't keep a registry of all the UserSecretReferences created by a given Polaris instance. 

Or are you suggesting something different?

Reply:
Yeah, I thinking mixing static and nonstatic methods is fine (and an intended feature of the language, as opposed to e.g. Scala where we must separate statics into an object and nonstatics into a class). If it keeps the related logic together without making that class too huge, let's try it.

Reply:
Done.